pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractDispatcherSingleActiveConsumer.java

```diff
@@ -261,6 +262,7 @@ public synchronized boolean canUnsubscribe(Consumer consumer) {
     public CompletableFuture<Void> close(boolean disconnectConsumers,
                                          Optional<BrokerLookupData> assignedBrokerLookupData) {
         IS_CLOSED_UPDATER.set(this, TRUE);
+        getRateLimiter().ifPresent(DispatchRateLimiter::close);
```

Comment:
Hi Lari,
Do we have some CI failure cases related to `SubscriptionMessageDispatchThrottlingTest` ?
Also, I wonder why we close `RateLimiter`  here could solve the issue because subclasses of `AbstractDispatcherSingleActiveConsumer` don't call this `close`.
And if we change here, we can also delete `close` in PersistentDispatcherSingleActiveConsumer.

Reply:
Yes, this fails.

Reply:
The lack of closing dispatch rate limiter could be causing resource leaks on released versions. On master branch the rate limiter no longer holds resources so leaving it unclosed is not impactful on master branch itself.

Reply:
@Technoboy- it fixed the test, so this should be the correct thing to do. The test is in the flaky tests.

Reply:
Reported #21756 with the issue.